### DISCUSSION TITLE
First full development release

### DISCUSSION THREAD

**Discussion by cjbarrie** [cjbarrie/academictwitteR#58] [General]
Release includes full set of functions, vignettes, and documentation.

<hr /><em>This discussion was created from the release <a href='https://github.com/cjbarrie/academictwitteR/releases/tag/0.0.0.9001'>First full development release</a>.</em>

**Comment by SangJungKim** [comment]
Hello! I have been using this r-package, and run into the error message.

So my code is:

tweets <- get_all_tweets("(Special gummies) OR Spliff OR Stoned OR Stoner OR thc OR (thc butter) OR (thc candy) OR (thc cookie) OR (thc cookie, ) OR (thc edible) OR Toking OR (Want a blunt) OR (want a joint) OR (want pot) OR Weed OR (Weed brownie) OR (weed butter) OR (Weed cake) OR (Weed candy) OR (weed edible) OR (weed gummies) OR (weed infused) OR (weed-infused)",
                         "2020-11-01T00:00:00Z",
                         "2020-11-30T23:59:59Z",
                         bearer_token = "MY BEARER TOKEN",
                         lang = "en",
                         data_path = "E:/2020-11/",
                         bind_tweets = FALSE)

and the error message returns:
Error: Argument 3 can't be a list containing data frames

Could you explain what might be the problem? Thanks!

**Reply by cjbarrie** [reply]
Hey! 

So we haven't incorporated grouping logics using parentheses yet. You could achieve this by just making a character vector of terms :

```
terms <- c("Special gummies", "Spliff", "thc butter")

```

Then just looping through with:

```

for(i in 1:length(terms)){
queryterm <- terms[[i]]

tweets <- get_all_tweets(term, ...)

}


```

**Reply by SangJungKim** [reply]
So I have followed your advice and revised the code:
terms <- c("Special gummies", "Spliff", "Stoned", "Stoner", "thc", "thc butter", "thc candy", "thc cookie", "thc cookie, ", "thc edible", "Toking", "Want a blunt", "want a joint", "want pot", "Weed", "Weed brownie", "weed butter", "Weed cake", "Weed candy", "weed edible", "weed gummies", "weed infused", "weed-infused")

for (i in 1:length(terms)){
+   queryterm <- terms[i]
+   tweets <- get_all_tweets(queryterm,
+                          "2020-11-01T00:00:00Z",
+                          "2020-11-30T23:59:59Z",
+                          bearer_token = "MY BEARER TOKEN",
+                          lang = "en",
+                          data_path = "E:/2020-11/",
+                          bind_tweets = FALSE)
+   }

but I am getting the same error: Error: Argument 1 can't be a list containing data frames

Could you debug this? Thank you!

**Reply by cjbarrie** [reply]
Hmmm... okay scratch my last reply as I just tried your original code and it works on my machine. Maybe @justinchuntingho has some suggestions?

**Reply by justinchuntingho** [reply]
The code works on my machine too. I did change the data_path since I am using a Mac, which makes me wonder, can you try supplying a folder name instead of a full path (eg data_path = "2020-11"). It should create a new folder under the working directory. If this is the problem, we might need a fix to handle windows absolute path.

**Reply by SangJungKim** [reply]
I tried that with my Windows machine, and I have run into the error again. I will try on mac and let you know if the code works or not!

**Reply by justinchuntingho** [reply]
I have done some digging and it seems the error message "Error: Argument 3 can't be a list containing data frames" comes  from the bind_rows() function of dplyr. I am wondering what version of dplyr have you installed? I tested on dplyr 1.0.6 it works for me. Perhaps try updating dplyr and see if it is the problem?
